lib/provisioner/worker/plugins/automators/chef_solo_automator/chef_solo_automator.rb

```diff
@@ -294,8 +280,8 @@ def install_chef(inputmap)
     # do we need sudo bash?
     sudo = 'sudo -E' unless sshauth['user'] == 'root'
 
-    write_ssh_file
-    @ssh_file = @task['config']['ssh-auth']['identityfile'] unless @ssh_keyfile.nil?
+    @ssh_keyfile = @task['config']['provider']['provisioner']['ssh_keyfile']
```

Comment:
``@ssh_keyfile`` can be replaced altogether with an ``unless``.  Or better yet, we already pass the ``@task`` object to ``write_ssh_file``, so why not just do the nil check there (which it sorta already does)?

Currently, if we removed this nil check before calling the function, ``write_ssh_file`` would still do the same check, do nothing as a result, but still return some path.  As a common function, it should handle the errors and have a clear API.